### PR TITLE
Legends arceus pokemon expansion

### PR DESCRIPTION
**Pull request by Blackforest92**
Add Legends Arceus pokemons to the expansion.

- Missing TM and Tutor moves.
- Female Basculegion icon seems to not showing.
- Missing some of battle sprites caused by unavailability.

--- **Commit updated in 19-August-2022**
- Add PLA data to `src/data/pokemon/form_species_tables.h` and `src/data/pokemon/form_species_tables_pointers.h`
- Add battle sprites for Kleavor

--- **Commit updated in 24-August-2022**
- Add Enamorus data to `src/date/pokemon/form_change_table_pointers.h` and `src/date/pokemon/form_change_tables.h`
- Pull latest `master` branch and fix merge conflicts

### Battle Sprites Credits
Kleavor			: leParagon
Ursaluna			: KingOfThe-X-Roads (frontsprite) & Asria (backsprite)
Overqwil			: KingOfThe-X-Roads (frontsprite) & AxelLoquendo (backsprite)
Growlithe-H		: KingOfThe-X-Roads (frontsprite) & Vent (backsprite)
Arcanine-H		: GriloKapu
Voltorb-H		: leParagon 
Electrode-H		: KingOfThe-X-Roads (frontsprite) & Vent (backsprite)
Typhlosion-H		: KingOfThe-X-Roads (frontsprite) & AxelLoquendo (backsprite)
Qwilfish-H		: Frander04
Sneasel-H		: Frander04
Lilligant-H		: KingOfThe-X-Roads (frontsprite) & AxelLoquendo (backsprite)
Zorua-H			: KingOfThe-X-Roads (frontsprite) & Lennybitao (backsprite)
Zoroark-H		: KingOfThe-X-Roads (frontsprite) & Lennybitao (backsprite)
Braviary-H		: KingOfThe-X-Roads (frontsprite) & AxelLoquendo (backsprite)
Sliggoo-H		: KingOfThe-X-Roads (frontsprite) & AxelLoquendo (backsprite)
Goodra-H		: KingOfThe-X-Roads (frontsprite) & AxelLoquendo (backsprite)
Avalugg-H		: KingOfThe-X-Roads (frontsprite) & AlexandreV2.0 (backsprite)
Basculin White	: KingOfThe-X-Roads (frontsprite) & AxelLoquendo (backsprite)
Dialga Origin		: KingOfThe-X-Roads (frontsprite) & Vent (backsprite)
Palkia Origin		: KingOfThe-X-Roads (frontsprite) & Vent (backsprite)

### CODE REVIEW

**Review by LOuroboros — CHANGES_REQUESTED** [review]
I remembered this PR exists now that I had to bring up the existence of the species form arrays, and now that I'm checking, this PR does not update them.
`SPECIES_BASCULIN_WHITE_STRIPPED` should be added to `sBasculinFormSpeciesIdTable` in `src/data/pokemon/form_species_tables.h`, for example.

**Comment by Blackforest92** [comment]
Added PLA data to `src/data/pokemon/form_species_tables.h` and `src/data/pokemon/form_species_tables_pointers.h`.

**Comment by Blackforest92** [comment]
Commit has been updated to latest `master` branch.
All merge conflicts have to do with `#if P_NEW_POKEMON == TRUE`
> Adjust `#endif` to include new pokemons

and [PR #2206](https://github.com/rh-hideout/pokeemerald-expansion/pull/2206) 
>Specifically tidying megas to their respective pokemon

Conflicts presented in : 
> 


> include/graphics.h 
include/constants/species.h 
(Adjust `#define FORMS_START SPECIES_CALYREX` to `#define FORMS_START SPECIES_ENAMORUS`)
include/constants/pokedex.h 
(Adjust `#define NATIONAL_DEX_COUNT  NATIONAL_DEX_CALYREX` to `#define NATIONAL_DEX_COUNT  NATIONAL_DEX_ENAMORUS`)
src/pokemon_icon.c
src/data/pokemon_graphics/front_pic_table.h
src/data/pokemon_graphics/back_pic_table.h
src/data/pokemon_graphics/palette_table.h
src/data/pokemon_graphics/shiny_palette_table.h
src/data/pokemon_graphics/enemy_mon_elevation.h
src/data/pokemon/base_stats.h
src/data/pokemon/pokedex_text.h
src/data/pokemon/pokedex_entries.h
src/data/pokemon/level_up_learnsets.h
src/data/pokemon/from_species_table_pointers.h
src/data/graphics/pokemon.h

**Comment by AsparagusEduardo** [comment]
It's very possible that the issue is with the debug menu, as current master has the same issue with other female icons, but not in the party menu or boxes.
image
To confirm that the icon is working, please give yourself a female Basculegion to test.

**Comment by Blackforest92** [comment]
[Basculegion_F.webm](https://user-images.githubusercontent.com/64682409/186649822-f192919a-490b-4e0f-828a-391f4f25b9d2.webm)
Tested quickly by adjusting gender ratio to 100% female,

File: src/data/pokemon/pokedex_text.h [hunk 1]
```diff
@@ -5349,4 +5349,44 @@ const u8 gCalyrexPokedexText[] = _(
...
+    "The black orbs shine with an uncanny light\n"
+    "when it is erecting invisible barriers.\n"
+    "The fur shed from its beard retains heat\n"
+    "and useful for winter clothing.");
+
+const u8 gKleavorPokedexText[] = _(
+    "A violent creature that fells trees with\n"
+    "its crude axes and shields itself with hard\n"
+    "stone. If one should encounter this Pokémon\n"
+    "in the wild, one's only recourse is to flee.");
```
Comment (AsparagusEduardo) [inline src/data/pokemon/pokedex_text.h:5363]:
```suggestion
    "A violent creature that fells trees with\n"
    "its crude axes and shields itself with hard\n"
    "stone. Should one encounter this Pokémon\n"
    "in the wild, one's only recourse is to flee.");
```

File: src/data/pokemon/pokedex_text.h [hunk 2]
```diff
@@ -5349,4 +5349,44 @@ const u8 gCalyrexPokedexText[] = _(
...
+    "Because of its wicked poison and daunting\n"
+    "physical prowess, no other species could\n"
+    "hope to best it on frozen highlands. It\n"
+    "prefers solitude and doesn't form packs.");
+
+const u8 gOverqwilPokedexText[] = _(
+    "Its lancelike spikes and savage\n"
+    "temperament have earned it the\n"
+    "nickname Sea Fiend.\n"
+    "It surps up poison to nourish itself.");
```
Comment (AsparagusEduardo) [inline src/data/pokemon/pokedex_text.h:5385]:
```suggestion
    "nickname 'sea fiend'.\n"
    "It slurps up poison to nourish itself.");
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: src/data/pokemon/pokedex_text.h (hunk 2 above)
Comment (LOuroboros) [reply src/data/pokemon/pokedex_text.h:5385]: Should be "“Sea Fiend.”" not 'sea fiend' like in other existing descriptions such as Landorus' imo.

Comment (AsparagusEduardo) [reply src/data/pokemon/pokedex_text.h:5385]: Forgot how to do quotes, fixed. But I chose lower case, as that's how it is in its entry in PLA. 
IMG_20220827_075025_783

Comment (LOuroboros) [reply src/data/pokemon/pokedex_text.h:5385]: LA didn't have to worry about keeping consistency being kind of detached from the mainline games and championing entirely new descriptions, so I stand by "Sea Fiend" with smart quotes for consistency with the other dex entries :eyes:

Comment (AsparagusEduardo) [reply src/data/pokemon/pokedex_text.h:5385]: Fair. Updated.

**Comment by Blackforest92** [comment]
Done.
Though i can't shake the weird feeling of adding another region names or any region names at all into a pokedex entry.
Its like still using a profile picture of you and your ex together when you are already stop dating.

**Comment by AsparagusEduardo** [comment]
Looks good, just needs to resolve the conflict with #2233. I can add the teachable data later.

**Comment by Blackforest92** [comment]
Done.

File: src/data/text/species_names.h [hunk 3]
```diff
@@ -1068,7 +1093,9 @@ const u8 gSpeciesNames[][POKEMON_NAME_LENGTH + 1] = {
     [SPECIES_ROTOM_FROST] = _("Rotom"),
     [SPECIES_ROTOM_MOW] = _("Rotom"),
     [SPECIES_ROTOM_WASH] = _("Rotom"),
-    // Giratina
+	// Origin
```
Comment (AsparagusEduardo) [inline src/data/text/species_names.h:1096]:
```suggestion
    // Origin
```

**Review by AsparagusEduardo — DISMISSED** [review]

File: src/data/text/species_names.h (hunk 3 above)
Comment (Blackforest92) [reply src/data/text/species_names.h]: // Origin Forme instead?
   Everything else is written as Origin Forme. i think

Comment (AsparagusEduardo) [reply src/data/text/species_names.h]: Sure.

**Comment by Blackforest92** [comment]
Did a `git reset --hard 61c33f5`, pull, then did the changes, then push.

**Pull request merged**